### DISCUSSION TITLE
Classes only used in @apply end up in resulting css

### DISCUSSION THREAD

**Discussion by tkalmar** [tailwindlabs/tailwindcss#10529] [Help] [answered]
Hi,
i'm using tailwind in an vue3 project with vite as build tool. It's working great so far but i have a question:
styles only used in @apply directly in the stlye block of my vue files don't  get removed from the resulting css.
Am i doing something wrong (besides overusing @apply) or is this a bug? How can i get rid of these unneded/unwanted styles?

**Comment by adamwathan** [comment, marked as answer]
Hey! Tailwind is picking up the class names you're using with `@apply` in your style blocks. If the false positives are really important to you, you'll want to use a custom `transform` to avoid scanning that part of each `.vue` file:

https://github.com/tailwindlabs/tailwindcss/discussions/10044

**Reply by tkalmar** [reply]
@adamwathan  Works like a charm, thanks. Is there  a reason to consider `@apply` blocks for class names?

**Reply by adamwathan** [reply]
Tailwind isn't really going out of its way to look for them — Tailwind is just very simple and scans the whole file indiscriminately, so it has no idea that you are using those classes with `@apply`, it just sees a string like `bg-black` and says "okay I better generate that class".

Can read more about how it works here:

https://tailwindcss.com/docs/content-configuration#class-detection-in-depth
